### ISSUE TITLE
tf.keras.layers.MultiHeadAttention causes a crash in tf.keras.layers.Softmax

### ISSUE THREAD

**Issue by schmackofatzen** [tensorflow/tensorflow#58322]
While working with the Multi Head Attention Layer I encountered a problem which seems to be an issue of the interplay between **tf.keras.layers.MultiHeadAttention** and  **tf.keras.layers.Softmax**.

Please check the following code piece to reproduce the error (which can be run standalone in a notebook):

```
import tensorflow as tf

start_number = [2.0]
output_array = tf.TensorArray(dtype=tf.float32, size=0, dynamic_size=True)
output_array = output_array.write(0, start_number)
x = tf.transpose(output_array.stack())

print("x:", x)

num_heads = 1
d_model = 1
dropout_rate = 0
mha = tf.keras.layers.MultiHeadAttention(num_heads=num_heads, key_dim=d_model, dropout=dropout_rate)
mha(query=x, key=x, value=x, use_causal_mask=True)
```

It will crash and the issue, in my opinion, is that in **def _build_attention(self, rank):** in **tf.keras.layers.MultiHeadAttention** creates a wrong input for **tf.keras.layers.Softmax**.

In **tf.keras.layers.MultiHeadAttention**  at **self._softmax = activation.Softmax(axis=norm_axes)**  the value for _norm_axis_ will be _"()"_.

This will then later in **tf.keras.layers.Softmax** in **def call(self, inputs, mask=None):** cause the calculation of the softmax to crash at **return backend.softmax(inputs, axis=self.axis[0])**

Mac OS Montery
Python 3.8.14
Tensorflow Version 2.10.0 (installed from binary including GPU acceleration)

**Comment by bhack** [comment]
As this is a Keras API you need to use the Keras repository:

https://github.com/keras-team/keras

**Comment by tiruk007** [comment]
@schmackofatzen 
As per above [comment ](https://github.com/tensorflow/tensorflow/issues/58322#issuecomment-1292196438). This issue seems to be a keras issue. Could you please post the issue on [Keras repo](https://github.com/keras-team/keras/issues).
Thank you!
